tests/cli/functional/actions/test_select_config_file.py

```diff
@@ -0,0 +1,115 @@
+                                               command=config_class.NAME)
+    assert message in output
+
+
+def test_auto_select_config_file(test_emitter,
+                                 stdout_trap,
+                                 alice_blockchain_test_config,
+                                 tmpdir,
+                                 mock_click_prompt):
+    """Only one configuration was found, so it was chosen automatically"""
+
+    config_class = alice_blockchain_test_config
+    config_path = Path(tmpdir) / config_class.generate_filename()
+
+    # Make one configuration
+    config_class.to_configuration_file(filepath=config_path)
+    assert config_path.exists()
+
+    result = select_config_file(emitter=test_emitter,
+                                config_class=config_class,
+                                config_root=tmpdir)
+
+    # ... ensure the correct account was selected
+    assert result == str(config_path)
+
+    # ... the user was *not* prompted
+    mock_click_prompt.assert_not_called()
+
+    # ...nothing was displayed
+    output = stdout_trap.getvalue()
+    assert not output
+
+
+def test_interactive_select_config_file(test_emitter,
+                                        stdout_trap,
+                                        alice_blockchain_test_config,
+                                        tmpdir,
+                                        mock_click_prompt,
+                                        mock_accounts,
+                                        patch_keystore):
+
+    """Multiple configurations found - Prompt the user for a selection"""
+
+    user_input = 0
+    config = alice_blockchain_test_config
+    config_class = config.__class__
+
+    # Make one configuration...
+    config_path = Path(tmpdir) / config_class.generate_filename()
+    config.to_configuration_file(filepath=config_path)
+    assert config_path.exists()
+    select_config_file(emitter=test_emitter,
+                       config_class=config_class,
+                       config_root=tmpdir)
+
+    # ... and then a bunch more
+    accounts = list(mock_accounts.items())
+    filenames = dict()
+    for filename, account in accounts:
+        config.checksum_address = account.address
+        config_path = Path(tmpdir) / config.generate_filename(modifier=account.address)
+        path = config.to_configuration_file(filepath=config_path, modifier=account.address)
+        filenames[path] = account.address
+        assert config_path.exists()
+
+    mock_click_prompt.return_value = user_input
+    result = select_config_file(emitter=test_emitter,
+                                config_class=config_class,
+                                config_root=tmpdir)
+
+    output = stdout_trap.getvalue()
+    for filename, account in accounts:
+        assert account.address in output
+
+    table_data = output.split('\n')
+    table_addresses = [row.split()[1] for row in table_data[2:-2]]
+
+    # TODO: Finish this test
```

Comment:
???

Reply:
I hear ya - When is this test done?  

Perhaps I can make a follow up Issue.  Unsure how granular we need to be around the GUI test at this time.